# Grouped Comparisons
Oliver, Kate, Mark

In [ ]:
import pandas as pd
import numpy as np

url = 'https://eds-217-essential-python.github.io/data/marine_microplastics.csv'
df = pd.read_csv(url)

df = df.dropna(subset=['Measurement'])
samples = df[df['Unit'] == 'pieces/m3'].copy()
samples['ocean'] = samples['Oceans'].str.replace(' Ocean', '')
positive = samples[samples['Measurement'] > 0].copy()

positive.shape

(7091, 23)

In [3]:
positive.groupby('ocean')['Measurement'].mean().sort_values(ascending = False)

ocean
Pacific     595.199821
Atlantic    282.729748
Arctic        2.279608
Southern      0.075703
Name: Measurement, dtype: float64

In [4]:
positive.groupby('ocean')['Measurement'].median().sort_values(ascending = False)

ocean
Pacific     0.116665
Arctic      0.044000
Atlantic    0.019440
Southern    0.011500
Name: Measurement, dtype: float64

In [5]:
positive['ocean'].value_counts()

ocean
Atlantic    6216
Pacific      799
Arctic        58
Southern      18
Name: count, dtype: int64

Southern and arctic oceans have too few samples to count

In [6]:
positive.groupby('ocean')['Measurement'].agg(['count', 'median', 'mean', 'max'])


,count,median,mean,max
ocean,,,,
Arctic,58,0.044000,2.279608,63.000000
Atlantic,6216,0.019440,282.729748,110480.000000
Pacific,799,0.116665,595.199821,21156.558533
Southern,18,0.011500,0.075703,0.499663


The large maximum for the atlantic ocean definitely skews the mean to be larger.  Therefore I would trust median more.

In [7]:
positive.groupby('ocean').agg({'Measurement': 'mean', 'Sampling Method': 'count', 'Organization': 'count'}) 

,Measurement,Sampling Method,Organization
ocean,,,
Arctic,2.279608,58,58
Atlantic,282.729748,6216,6216
Pacific,595.199821,799,799
Southern,0.075703,18,18


In [8]:
positive.groupby('Sampling Method')['Measurement'].agg(['count', 'median']).sort_values('median', ascending = False)


,count,median
Sampling Method,,
Stainless steel spoon,50,21840.000000
Van Dorn sampler,84,2000.000000
Aluminum bucket,57,1710.000000
PVC cylinder,292,1410.437236
CTD rosette sampler,4,21.000000
Intake seawater pump,115,2.500000
Plankton net,74,0.840000
Hand picking,12,0.777778
Manta net,884,0.252943


Largest median: stainless steel spoon (21840)
Smallest median: grab sample (0.0060)


These are separated by six orders of magnitude

If the oceans were not sampled with the same mix of sampling methods, the means won't really be comparable since different methods have different efficiencies and introduce different biases.

In [9]:
by_measurement = positive.groupby(['ocean', 'Sampling Method'])['Sampling Method'].value_counts().sort_values(ascending = False)
by_measurement.reset_index()

,ocean,Sampling Method,count
0,Atlantic,Neuston net,4390
1,Atlantic,Grab sample,748
2,Atlantic,Manta net,631
3,Pacific,Manta net,226
4,Pacific,PVC cylinder,188
5,Pacific,Grab sample,166
6,Pacific,Neuston net,158
7,Atlantic,Intake seawater pump,106
8,Atlantic,PVC cylinder,104
9,Atlantic,Van Dorn sampler,84


In [17]:
atlantic = positive[positive['ocean'] == 'Atlantic'].copy()
pacific = positive[positive['ocean'] == 'Pacific'].copy()

print("Atlantic")
print(atlantic.value_counts('Sampling Method'))

print("Pacific")
print(pacific.value_counts('Sampling Method'))

Atlantic
Sampling Method
Neuston net              4390
Grab sample               748
Manta net                 631
Intake seawater pump      106
PVC cylinder              104
Van Dorn sampler           84
Aluminum bucket            57
Stainless steel spoon      50
Plankton net               34
Hand picking               12
Name: count, dtype: int64
Pacific
Sampling Method
Manta net               226
PVC cylinder            188
Grab sample             166
Neuston net             158
Plankton net             40
AVANI net                18
Intake seawater pump      3
Name: count, dtype: int64


In [19]:
neuston_net = positive[positive['Sampling Method'] == 'Neuston net'].copy()

neuston_net.groupby('ocean')['Measurement'].agg(['count', 'median', 'mean'])

,count,median,mean
ocean,,,
Atlantic,4390,0.015120,0.085767
Pacific,158,0.030680,0.396195
Southern,4,0.306271,0.278664
